src/main/java/io/gingersnapproject/cdc/event/InternalEventManager.java

```diff
@@ -0,0 +1,53 @@
+package io.gingersnapproject.cdc.event;
+
+import java.net.URI;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+/**
+ * The global event manager.
+ * <p>Responsible for publishing events for stages of the engines for the different CDI {@link Event}.
+ * This approach is based on the <a href="https://github.com/debezium/debezium/blob/d42b5480a3965ca1e7761a5cc7a39af7c890ee2d/debezium-server/debezium-server-core/src/main/java/io/debezium/server/ConnectorLifecycle.java">Debezium Server</a>.</p>
+ *
+ * <p>Adopting this approach, multiple listeners with different goals can receive the events without
+ * changing the internals for specific reasons.</p>
+ *
+ * @author Jose Bolina
+ */
+@ApplicationScoped
+public class InternalEventManager {
```

Comment:
Can this just be `EventManager`?

Reply:
I've put the `Internal` to create some distinction between our events and Debezium's events. Think that's necessary?

Reply:
Argh I see what you mean now. We could always use `NotificationManager` like in the HR client to differentiate?

Or `LifeycleEventManager`?

I'm just trying to think of something that's a little more informative than "internal".

Reply:
`NotificationManager` sounds good to me.